source/sessions/driver-sessions.rst

```diff
@@ -315,6 +315,9 @@ This property returns the session ID of this session. Note that if server
 sessions are pooled, different ``ClientSession`` instances will have the same session ID,
 but never at the same time.
 
+After a ``ClientSession`` has ended, accessing ``sessionId`` MUST either return
+a null value or raise an exception.
+
```

Comment:
I disagree.

The application may wish to see what session Id was used even after the session itself has been ended.

Reply:
Even though the session id is going to be reused by another ClientSession? I think of ClientSession.sessionId as a delegated property that really belongs to the ClientSession's ServerSession: once the ServerSession is back in the session pool, a ClientSession should have no sessionId.

Reply:
I see what you are saying... but an application writer may not understand that subtlety.

From their point of view they simply want to find out what the session ID is, and it might surprise them to find that it has mysteriously become null or unavailable.

Reply:
I see parallels with libmongoc's cursor ID accessor returning `0` once a cursor has closed, while a live cursor would return a non-zero value. Although I have no idea what other drivers do in that situation, I expect most would treat these cases similarly (and want to use the same convention for each).